src/Boids.hs

```diff
@@ -149,10 +156,31 @@ initialState :: IO World
 initialState = do boids <- replicateM 200 randomBoid
                   return $ World { worldBoids = boids
                                  , worldGuide = (0.0, 0.0)
+                                 , worldViewPort = viewPortInit
+                                 , worldDraggingPivot = Nothing
                                  }
 
+zoom :: Float -> ViewPort -> ViewPort
+zoom step viewPort = viewPort {
+                       viewPortScale = max (scale + step) 0.0
+                     }
+    where scale = viewPortScale viewPort
+
+handleInput :: Event -> World -> World
+
+-- TODO: take cursor position into account during zooming
+handleInput (EventKey (MouseButton WheelUp) Down _ _) world =
+    world { worldViewPort = zoom zoomSpeed $ worldViewPort world }
+handleInput (EventKey (MouseButton WheelDown) Down _ _) world =
+    world { worldViewPort = zoom (-zoomSpeed) $ worldViewPort world }
+-- TODO: implement dragging around
+-- TODO: implement boids following the mouse cursor
+handleInput _ world = world
+
 renderState :: World -> Picture
-renderState = pictures . map renderBoid . worldBoids
+renderState world = applyViewPortToPicture viewPort frame
```

Comment:
What be great to unit test that as well, but that might be difficult. See what we can do here.